c10/util/BFloat16-math.h

```diff
@@ -91,4 +91,58 @@ inline c10::BFloat16 fmod(c10::BFloat16 a, c10::BFloat16 b) {
   return std::fmod(float(a), float(b));
 }
 
+C10_HOST_DEVICE inline c10::BFloat16 nextafter(
+    c10::BFloat16 from,
+    c10::BFloat16 to) {
+  // inspired from :
+  // https://github.com/ifduyue/musl/blob/master/src/math/nextafter.c
+  using int_repr_t = uint16_t;
+  using float_t = c10::BFloat16;
+  constexpr uint8_t bits = 16;
+  union {
+    float_t f;
+    int_repr_t i;
+  } ufrom = {from}, uto = {to};
+
+  // get a mask to get the sign bit i.e. MSB
+  int_repr_t sign_mask = int_repr_t{1} << (bits - 1);
+
+#if defined(__HIP_PLATFORM_HCC__) || defined(_MSC_VER)
+  if (from != from || to != to) {
+#else
+  if (std::isnan(from) || std::isnan(to)) {
+#endif
+    return from;
+  }
+
+  // if they are exactly the same.
```

Comment:
Nice short-circuit but the comment should explicitly state this is a short-circuit and the short-circuit criteria, like "Short-circuits if from and to are the same"